schema.prisma

```diff
@@ -0,0 +1,682 @@
+}
+
+model PracticeGroupTranslations {
+  language        Language
+  practiceGroupId String   @db.Uuid
+
+  comment String
+
+  practiceGroup PracticeGroup @relation(fields: [practiceGroupId], references: [id])
+
+  @@id([language, practiceGroupId])
+}
+
+// ########################################
+//  核心能力相關
+// ########################################
+
+// TODO: fix relations, translation model
+
+// model CourseCapability {
+//   /// 課程識別碼
+//   courseId String @db.VarChar(9)
+//   // 待確認是否需要
+//   // deparmentId String @db.VarChar(4)
+//   /// 建議修課年級，[1-7]
+//   recommendedGrades Int[]  
+// }
+
+/// 系所核心能力
+model DeparmentCapability {
+  /// 系所代碼
+  deparmentId    String @db.VarChar(4)
+  /// 核心代碼
+  capabilityCode String @db.VarChar(1)
+
+  @@unique([deparmentId, capabilityCode])
+}
+
+/// 核心能力類型
+model CapabilityType {
+  /// 核心代碼， [A-Z]
+  code        String @id @db.VarChar(1)
+  /// 中文名稱
+  nameZhTw    String @db.VarChar(100)
+  /// 英文名曾
+  nameEn      String @db.VarChar(4000)
+  /// 中文說明
+  commentZhTw String @db.VarChar(150)
+  /// 英文說明
+  commentEn   String @db.VarChar(4000)
+}
+
+/// 課程分類
+model CourseCustomType {
+  /// 課程分類代碼， [A-Z]
+  code     String @id @db.VarChar(1)
+  /// 中文名稱
+  nameZhTw String @db.VarChar(50)
+  /// 英文名曾
+  nameEn   String @db.VarChar(50)
+}
+
+// ########################################
+//  領域專長
+// ########################################
+
+/// 領域專長說明主表
+model Domain {
+  /// 領域專長代碼
+  code   String  @id @db.VarChar(10)
+  /// 名稱
+  name   String  @db.VarChar(50)
+  /// 英文名稱
+  nameEn String  @db.VarChar(100)
+  /// 是否通過
+  apply  Boolean
+}
+
+/// 課程領域專長關聯
+model CourseDomain {
+  /// 課程識別碼
+  CourseId   String @db.VarChar(9)
+  /// 領域專長代碼
+  DomainCode String @db.VarChar(10)
+
+  @@unique([CourseId, DomainCode])
+}
+
+// ########################################
+// 課程大綱 from ceiba db
+// ########################################
+
+// 有許多重複的欄位，待確認此用方法
+// 有許多重複的欄位，待確認此用方法
+// 有許多重複的欄位，待確認此用方法
+
+/// 課程大綱主表
+model Syllabus {
+  // 原sn
+  id       Int    @id
+  /// 學期，如103-1
+  semester String @db.VarChar(6)
+  /// 課程識別碼
+  // CEIBA 課程網頁有非正式課程及正式課程兩種。非正式課程指不在正式課程中的課程。
+  // 正式課程，
+  // 課程識別碼如 101 80002
+
+  // 非正式課程，
+  // 課程識別碼，前三碼為XXX 80002
+  courseNo       String                 @db.VarChar(10)
+  /// 班級
+  classNo        String                 @db.VarChar(5)
+  /// CEIBA 專用之 6 碼課號
+  // 開設 CEIBA 網頁之課程皆有唯一的 6 碼課號。
+  // 未開設 CEIBA 課程網頁者，沒有。
+  courseSn       String?                @db.VarChar(6)
+  // lang 欄位為big5 時，格式為 第1週
+  // lang 欄位為 eng 時，格式為 Week 1
+  unit           String                 @db.VarChar(30)
+  /// 本筆課程進度表的日期
+  note           String                 @db.VarChar(256)
+  /// 是否為異動或新增
+  // 匯入之前學期的課程資料時填入 0 ，只要有異動或是新增時都填入 1 。 
+  // 區分課程進度的建立類型用。
+  flag           Boolean
+  /// 課程主題
+  subject        String
+  /// 課程進度表允許顯示的日期中的年月日。
+  // 過了設定的期限才會顯示給同學看見
+  publicDate     DateTime
+  /// 表格內容類型
+  classification SyllabusClassification
```

Comment:
Will this be an ID of another table? If that will, please use `SyllabusClassificationId` instead.